test/SourceKit/InterfaceGen/gen_public_module_name.swift

```diff
@@ -0,0 +1,89 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t --leading-lines
+
+// RUN: %target-swift-frontend %t/common.swift \
+// RUN:   -module-name Common \
+// RUN:   -emit-module-path %t/Common.swiftmodule \
+// RUN:   -emit-module-interface-path %t/Common.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6
+
+// RUN: %target-swift-frontend %t/subimpl.swift -I %t \
+// RUN:   -module-name SubImpl \
+// RUN:   -emit-module-path %t/SubImpl.swiftmodule \
+// RUN:   -emit-module-interface-path %t/SubImpl.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6
+//
+// RUN: %target-swift-frontend %t/modimpl.swift -I %t \
+// RUN:   -module-name ModImpl \
+// RUN:   -emit-module-path %t/ModImpl.swiftmodule \
+// RUN:   -emit-module-interface-path %t/ModImpl.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6 \
+// RUN:   -public-module-name PubMod
+
+// RUN: %target-swift-frontend %t/mod.swift -I %t \
+// RUN:   -module-name PubMod \
+// RUN:   -emit-module-path %t/PubMod.swiftmodule \
+// RUN:   -emit-module-interface-path %t/PubMod.swiftinterface \
+// RUN:   -enable-library-evolution -swift-version 6
+
+// RUN: %sourcekitd-test -req=interface-gen -module PubMod -- -swift-version 6 -I %t -target %target-triple &> %t/PubMod.generatedinterface
+// RUN: %FileCheck -input-file=%t/PubMod.generatedinterface -check-prefix INTERFACE %s
+
+//--- common.swift
+public struct CommonType {}
+
+
+//--- subimpl.swift
+public struct SubImplTy {}
+
+
+//--- modimpl.swift
+import struct Common.CommonType
+import SubImpl
+
+public struct ImplTy {
+  public let member: SubImplTy
+  public let member2: SubImpl.SubImplTy
+}
+
+public func fromModImpl(arg: ImplTy, arg2: ModImpl.ImplTy) {}
+
+
+//--- mod.swift
+@_exported import ModImpl
+import struct Common.CommonType
+
+public struct ModTy {
+  public let member: ImplTy
+  public let member2: ModImpl.ImplTy
+}
+
+public func fromMod(arg: ModTy, arg2: PubMod.ModTy) {}
+
+// INTERFACE-NOT: import ModImpl
+// INTERFACE: import struct Common.CommonType
+// INTERFACE: import SubImpl
+// INTERFACE-NOT: import
+// INTERFACE: struct ImplTy
+// INTERFACE:   let member: SubImpl.SubImplTy
+// INTERFACE:   let member2: SubImpl.SubImplTy
+// INTERFACE: struct ModTy
+// INTERFACE:   let member: PubMod.ImplTy
+// INTERFACE:   let member2: PubMod.ImplTy
+// INTERFACE: func fromMod(arg: PubMod.ModTy, arg2: PubMod.ModTy)
+// INTERFACE: func fromModImpl(arg: PubMod.ImplTy, arg2: PubMod.ImplTy)
```

Comment:
If I read the source code correctly, these types are not module-qualified if the module of the type is part of the SDK. Is it easy to add another `interface-gen` test treating these non-user modules?

Reply:
Changed to non-user only 👍(and yeah, looks like there's no module qualification there)